controllers/config_map.go

```diff
@@ -215,24 +284,52 @@ func (r *ConfigMapReconciler) reconcileProviderConfig(ctx context.Context, clust
 	return nil
 }
 
-func (r *ConfigMapReconciler) reconcileDelete(ctx context.Context, cluster *capa.AWSCluster) (ctrl.Result, error) {
+func (r *ConfigMapReconciler) updateProviderConfig(ctx context.Context, providerConfig *unstructured.Unstructured, accountID string) error {
 	logger := log.FromContext(ctx)
-	logger.Info("Reconcile delete")
-	defer logger.Info("Done deleting")
 
-	config := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      fmt.Sprintf("%s-crossplane-config", cluster.Name),
-			Namespace: cluster.Namespace,
+	patchedConfig := providerConfig.DeepCopy()
+	patchedConfig.Object["spec"] = r.getProviderConfigSpec(accountID)
+	err := r.Client.Patch(ctx, patchedConfig, client.MergeFrom(providerConfig))
+	if err != nil {
+		logger.Error(err, "Failed to patch provider config")
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func (r *ConfigMapReconciler) getProviderConfigSpec(accountID string) map[string]interface{} {
+	return map[string]interface{}{
+		"credentials": map[string]interface{}{
+			"source": "WebIdentity",
+			"webIdentity": map[string]interface{}{
+				"roleARN": fmt.Sprintf("arn:aws:iam::%s:role/crossplane-assume-role", accountID),
```

Comment:
this will not work in china `aws` should be variable based on region and if region is prefixed with `cn` it should be `aws-cn`

Also I do not like one static role for everything, we should at least prepend it with cluster/mc name to  be able to determine  where is it used.

If cross-plane will be more spread and customer will start using as well we should have isolated roles per cluster for sure

Reply:
Done, now supports china and I've put both roles as parameters.